lib/Auth/Process/LogUser.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+use Sil\SspUtils\Metadata;
+
+use Aws\DynamoDb\Exception\DynamoDbException;
+use Aws\DynamoDb\Marshaler;
+
+/**
+ * This Auth Proc logs information about each successful login to an AWS Dynamodb table.
+ *
+ *  It requires the following configs
+ *   'DynamoEndpoint' ex. http://arn:aws:dynamodb:us-east-1:1234567
+ *   'DynamoRegion' ex. 'us-east-1'
+ *   'DynamoLogTable' ex. 'sildisco_dev_user-log'
+ *
+ */
+class sspmod_sildisco_Auth_Process_LogUser extends SimpleSAML_Auth_ProcessingFilter
+{
+    const IDP_KEY = "saml:sp:IdP"; // the key that points to the entity id in the state
+
+    // the metadata key for the IDP's Namespace code (i.e. short name)
+    const IDP_CODE_KEY = 'IDPNamespace';
+
+    const DYNAMO_ENDPOINT_KEY = 'DynamoEndpoint';
+
+    const DYNAMO_REGION_KEY = 'DynamoRegion';
+
+    const DYNAMO_LOG_TABLE_KEY = 'DynamoLogTable';
+
+
+    // The host of the aws dynamodb
+    private $dynamoEndpoint;
+
+    // The region of the aws dynamodb
+    private $dynamoRegion;
+
+    // The name of the aws dynamodb table that stores the login data
+    private $dynamoLogTable;
+
+    /**
+     * Initialize this filter, parse configuration.
+     *
+     * @param array $config  Configuration information about this filter.
+     * @param mixed $reserved  For future use.
+     */
+    public function __construct($config, $reserved) {
+        parent::__construct($config, $reserved);
+        assert(is_array($config));
+
+        if (! empty($config[self::DYNAMO_ENDPOINT_KEY])) {
+            $this->dynamoEndpoint = $config[self::DYNAMO_ENDPOINT_KEY];
+        }
+
+        if (! empty($config[self::DYNAMO_REGION_KEY])) {
+            $this->dynamoRegion = $config[self::DYNAMO_REGION_KEY];
+        }
+
+        if (! empty($config[self::DYNAMO_LOG_TABLE_KEY])) {
+            $this->dynamoLogTable = $config[self::DYNAMO_LOG_TABLE_KEY];
+        }
+
+    }
+
+    /**
+     * Apply filter to copy attributes.
+     *
+     * @param array &$state  The current state array
+     */
+    public function process(&$state) {
+        if (! $this->configsAreValid()) {
+            return;
+        }
+
+        assert('is_array($state)');
```

Comment:
Using a string in the `assert` now triggers a PHP warning, so you'll probably want to change it from `assert('is_array($state)');` to `assert(is_array($state));`

Reply:
Fixed